DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/README.md

```diff
@@ -0,0 +1,392 @@
+
+
+# Register Map and Streaming Interfaces
+This FPGA tutorial demonstrates how to specify the kernel invocation interfaces and kernel argument interfaces, and demonstrates the differences between streaming interfaces that use a ready/valid handshake, and register-mapped interfaces that exist in the kernel's control/status register (CSR).
+
+The kernel invocation interface (namely, the `start` and `done` signals) can be implemented in the kernel's CSR, or using a ready/valid handshake. Similarly, the kernel arguments can be passed through the CSR, or through dedicated conduits. The invocation interface and any argument interfaces are specified independently, so you may choose to implement the invocation interface with a ready/valid handshake, and implement the kernel arguments in the CSR. All argument interfaces that are implemented as conduits will be synchronized to the ready/valid handshake of the kernel invocation interface. This means that it is not possible to configure a kernel with a register-mapped invocation interface and conduit arguments. The following table lists valid kernel argument interface synchronizations.
+
+| Invocation Interface    | Argument Interface    | Argument Interface Synchronization       |
+|----------------------|-----------------------|------------------------------------------|
+| Streaming            | Streaming             | Synchronized with `start` and `ready_out` |
+| Streaming            | Register mapped       | N/A                                      |
+| Register mapped      | Streaming             | *No synchronization possible*            |
+| Register mapped      | Register mapped       | N/A                                      |
+
+> **Note**: Register mapped kernel arguments are not currently supported in kernels with a streaming invocation interface.
+
+If you would like an argument to have its own dedicated ready/valid handshake, please implement that argument using a [Host Pipe](../hostpipes/).
+
+> **Note**: The register map and streaming interface feature is only supported in the IP Component Authoring design flow. The IP Component Authoring design flow compiles SYCL* source code to standalone IPs that can be deployed into your Intel® Quartus® Prime projects. The generated IP is not meant to run on FPGA devices directly, therefore there will be no FPGA executables generated in this tutorial. Emulator and simulator executables are still generated to allow you to validate your IP. You can run the generated HDL through Intel® Quartus® Prime to generate accurate f<sub>MAX</sub> and area estimates.
+
+***Documentation***:  The [DPC++ FPGA Code Samples Guide](https://software.intel.com/content/www/us/en/develop/articles/explore-dpcpp-through-intel-fpga-code-samples.html) helps you to navigate the samples and build your knowledge of DPC++ for FPGA. <br>
```

Comment:
Remove the Documentation section entirely as well as everything below up to the Purpose section